src/experiments.js

```diff
@@ -161,21 +153,16 @@ function calcExperimentOn(win, experimentId) {
  */
 export function toggleExperiment(win, experimentId, opt_on,
     opt_transientExperiment) {
-  const toggles = experimentToggles();
-  const experimentIds = getExperimentIds(win);
-  const currentlyOn = (experimentIds.indexOf(experimentId) != -1) ||
-      (experimentId in toggles && toggles[experimentId]);
-  const on = opt_on !== undefined ? opt_on : !currentlyOn;
+  const currentlyOn = isExperimentOn(win, experimentId);
+  const on = !!(opt_on !== undefined ? opt_on : !currentlyOn);
   if (on != currentlyOn) {
-    if (on) {
-      experimentIds.push(experimentId);
-      toggles[experimentId] = true;
-    } else {
-      experimentIds.splice(experimentIds.indexOf(experimentId), 1);
-      toggles[experimentId] = false;
-    }
+    const toggles = experimentToggles();
+    toggles[experimentId] = on;
+
+    const cookieToggles = getExperimentTogglesFromCookie(win);
+    cookieToggles[experimentId] = on;
```

Comment:
This means `opt_transientExperiment` stores could be persisted if anything after it persists.